### PR TITLE
CLI: info: handle durations > 2**63

### LINKED ISSUE #822: negative end time reported by `mcap info`, disagrees with duration
**Description**

Repro file: [zfp.mcap.zip](https://github.com/foxglove/mcap/files/10773480/zfp.mcap.zip)

```
$ mcap info zfp.mcap
library: libmcap 1.0.0
profile: 
messages: 300
duration: 2562047h47m16.854775807s
start: 0.000
end: -0.090
compression:
        zstd: [1/1 chunks] [221.93 KiB/3.77 KiB (98.30%)] [0.00 B/sec] 
channels:
        (1) image              100 msgs (0.00 Hz)   : foxglove.CompressedImage [protobuf]    
        (2) camera_info        100 msgs (0.00 Hz)   : foxglove.CameraCalibration [protobuf]  
        (3) image_annotations  100 msgs (0.00 Hz)   : foxglove.ImageAnnotations [protobuf]   
attachments: 0
metadata: 0
```

### PR DESCRIPTION
**Pull request by james-rms**
### Public-Facing Changes
`mcap info` on files with a duration > 2**63s will have their duration, start and end times printed correctly.

### Description
* fix a bug in decimalTime which would print timestamps > 2^63ns as negative numbers
* fix a bug where very long durations in MCAP would be printed as negative.

For the linked file in #822, returns:
```
$ mcap info
library: libmcap 1.0.0
profile: 
messages: 300
duration: 18446744073.620s
start: 0.000000000
end: 18446744073.619617024
compression:
        zstd: [1/1 chunks] [221.93 KiB/3.77 KiB (98.30%)] [0.00 B/sec] 
channels:
        (1) image              100 msgs (0.00 Hz)   : foxglove.CompressedImage [protobuf]    
        (2) camera_info        100 msgs (0.00 Hz)   : foxglove.CameraCalibration [protobuf]  
        (3) image_annotations  100 msgs (0.00 Hz)   : foxglove.ImageAnnotations [protobuf]   
attachments: 0
metadata: 0
```

Fixes #822

### CODE REVIEW

**Comment by james-rms** [comment]
cc. @jhurliman

File: go/cli/mcap/cmd/info.go [hunk 1]
```diff
@@ -45,16 +45,23 @@ func printInfo(w io.Writer, info *mcap.Info) error {
 		fmt.Fprintf(buf, "messages: %d\n", info.Statistics.MessageCount)
 		start = info.Statistics.MessageStartTime
 		end = info.Statistics.MessageEndTime
+		durationNs := float64(end) - float64(start)
+		durationInSeconds = durationNs / 1e9
```
Comment (jtbandes) [inline go/cli/mcap/cmd/info.go:49]: Can we do the subtraction before converting to float? That would be less likely to lose precision due to catastrophic cancellation.

Comment (james-rms) [reply go/cli/mcap/cmd/info.go:49]: yes, but it's a little more complex - we have to make sure the unsigned subtraction won't overflow first.

Comment (jtbandes) [reply go/cli/mcap/cmd/info.go:49]: sounds good...maybe an error message on negative duration? or do we want to actually display it with a `-` sign?

File: go/cli/mcap/cmd/info.go [hunk 2]
```diff
@@ -35,6 +35,19 @@ func humanBytes(numBytes uint64) string {
...
+	// float64 can represent the entire range of possible durations (-2**64, 2**64),
+	// albeit with some loss of precision.
+	diff := end - start
+	signMultiplier := 1.0
+	if start > end {
+		diff = start - end
+		signMultiplier = -1.0
+	}
+	return float64(diff) * signMultiplier
+}
```
Comment (jtbandes) [inline go/cli/mcap/cmd/info.go:49]: could we add a few quick unit tests for this, or for the printInfo function to prove that it handles the edge cases we want it to handle?

**Review by jtbandes — APPROVED** [review]

File: go/cli/mcap/cmd/info.go (hunk 2 above)
Comment (james-rms) [reply go/cli/mcap/cmd/info.go:49]: yup, sure can.
